### PR TITLE
Feature/placeable items

### PR DESCRIPTION
**Pull request by plally**
Need models/icons for:
- [x] Grill
- [x] Water Purifier

Icons for:
- [x] Storage boxes (all 3)
- [x] Workbenches (all ?)

### CODE REVIEW

**Review by samuelWilliams99 — APPROVED** [review]
Good job my g u y

File: gamemode/modules/util/sh_models.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+local function entCreate( model )
+    local ent = ents.Create( "prop_physics" )
+    ent:SetModel( model )
+    
+    return ent
+end
+
+if CLIENT then entCreate = ents.CreateClientProp end
+
+
```
Comment (samuelWilliams99) [inline gamemode/modules/util/sh_models.lua]:
```suggestion

```

File: gamemode/modules/util/sh_models.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+local modelBoundsCache = {}
+
+function RVR.Util.GetModelBounds( model )
+    local cached = modelBoundsCache[model]
+    if cached then
+        return unpack(cached)
```
Comment (samuelWilliams99) [inline gamemode/modules/util/sh_models.lua]:
```suggestion
        return unpack( cached )
```

File: gamemode/modules/items/sh_items.lua [hunk 3]
```diff
@@ -92,6 +92,52 @@ items.items = {
...
+        model = "models/props_borealis/bluebarrel001.mdl", -- TODO a better model?
+        icon = "materials/rvr/items/water_bottle.png", -- TODO proper icon
+        placeable = true,
+        placeableClass = "rvr_water_purifier",
+        viewModelOffset = Vector( 30, 50, -9 )
+    },
+    {
+        type = "grill",
+        displayName = "Grill",
+        description = "Cook your food. Or eat it raw...",
```
Comment (plally) [inline gamemode/modules/items/sh_items.lua]:
```suggestion
        description = "Cook your food. Or eat it, rawr~",
```
at @samuelWilliams99 request

File: gamemode/modules/items/sh_items.lua [hunk 4]
```diff
@@ -115,6 +115,29 @@ items.items = {
...
```
Comment (samuelWilliams99) [inline gamemode/modules/items/sh_items.lua]: rawr xd?

**Review by samuelWilliams99 — CHANGES_REQUESTED** [review]
Have you checked `GetModelBounds` works on client and server?
Also might wanna put all the models/icons required in a checklist in the PR description

**Review by samuelWilliams99 — APPROVED** [review]

File: entities/entities/rvr_crafter/init.lua [hunk 5]
```diff
@@ -23,6 +23,10 @@ function ENT:Initialize()
     RVR.Crafting.makeCrafter( self, self.CrafterName, self.Tier, self.CrafterType )
 end
 
+local nextUse = 0
```
Comment (overlordbleck) [inline entities/entities/rvr_crafter/init.lua:26]: Why dont we just assign this to a variable on the entity

File: gamemode/config/crafting.lua [hunk 6]
```diff
@@ -29,6 +29,37 @@ recipe = {
...
+            {
+                item = "water",
+                count = 1,
+                timeToCraft = 39,
+                ingredients = {
+                    dirty_water =  1
+                }
+            }
+        }
+    },
```
Comment (overlordbleck) [inline gamemode/config/crafting.lua:46]: Is this just craftable all the time? Or is it only craftable via the water purifier?

Comment (overlordbleck) [reply gamemode/config/crafting.lua:46]: Also, where is the recipe for the purifier?

File: gamemode/config/crafting.lua [hunk 7]
```diff
@@ -29,6 +29,37 @@ recipe = {
...
+        icon = "materials/rvr/items/tuna.png",
+        crafterType = "cooking",
+        recipes = {
+            {
+                item = "cooked_tuna",
+                count = 1,
+                timeToCraft = 5,
+                ingredients = {
+                    tuna = 1,
+                    wood = 1
```
Comment (overlordbleck) [inline gamemode/config/crafting.lua:58]: 🤔

File: entities/entities/rvr_crafter/init.lua (hunk 5 above)
Comment (plally) [reply entities/entities/rvr_crafter/init.lua:26]: its not used anywhere except in this single function

**Pull request merged**